src/System.Runtime/tests/System/Buffer.cs

```diff
@@ -4,111 +4,211 @@
+
+        Assert.Throws<ArgumentException>("array", () => Buffer.ByteLength(Array.CreateInstance(typeof(DateTime), 25))); // Array is not a primitive
+        Assert.Throws<ArgumentException>("array", () => Buffer.ByteLength(Array.CreateInstance(typeof(decimal), 25))); // Array is not a primitive
+        Assert.Throws<ArgumentException>("array", () => Buffer.ByteLength(Array.CreateInstance(typeof(string), 25))); // Array is not a primitive
+    }
+
+    [Theory]
+    [InlineData(new uint[] { 0x01234567, 0x89abcdef }, 0, 0x67)]
+    [InlineData(new uint[] { 0x01234567, 0x89abcdef }, 7, 0x89)]
+    public static void TestGetByte(Array array, int index, int expected)
+    {
+        Assert.Equal(expected, Buffer.GetByte(array, index));
+    }
+
+    [Fact]
+    public static void TestGetByte_Invalid()
+    {
+        var array = new uint[] { 0x01234567, 0x89abcdef };
+
+        Assert.Throws<ArgumentNullException>("array", () => Buffer.GetByte(null, 0)); // Array is null
+        Assert.Throws<ArgumentException>("array", () => Buffer.GetByte(new object[10], 0)); // Array is not a primitive array
+
+        Assert.Throws<ArgumentOutOfRangeException>("index", () => Buffer.GetByte(array, -1)); // Index < 0
+        Assert.Throws<ArgumentOutOfRangeException>("index", () => Buffer.GetByte(array, 8)); // Index >= array.Length
+    }
+
+    [Theory]
+    [InlineData(0, 0)]
+    [InlineData(0, 5000)]
+    [InlineData(5000, 6000)]
+    public static unsafe void TestMemoryCopy_Long(int sourceIndexOffset, int destinationIndexOffset)
+    {
+        var sourceArray = new int[25000];
+        for (int i = 0; i < sourceArray.Length; i++)
+        {
+            sourceArray[i] = i;
+        }
+
+        var destinationArray = new int[30000];
+        fixed (int* sourceBase = sourceArray, destinationBase = destinationArray)
+        {
+            Buffer.MemoryCopy(sourceBase + sourceIndexOffset, destinationBase + destinationIndexOffset, 30000 * 4, (25000 - sourceIndexOffset) * 4);
+        }
+
+        for (int i = sourceIndexOffset; i < sourceArray.Length; i++)
+        {
+            Assert.Equal(sourceArray[i], destinationArray[i + destinationIndexOffset - sourceIndexOffset]);
+        }
+    }
+
+    [Fact]
+    public static unsafe void TestMemoryCopy_OverlappingBuffers()
+    {
+        var array = new int[200];
+        for (int g = 0; g < array.Length; g++)
+        {
+            array[g] = g;
+        }
+
+        fixed (int* arrayBase = array)
         {
-            Assert.Equal(length * size, Buffer.ByteLength(array));
+            Buffer.MemoryCopy(arrayBase, arrayBase + 50, 200 * 4, 100 * 4);
         }
-        else
+
+        for (int g = 0; g < 100; g++)
         {
-            Assert.Throws<ArgumentException>(() => Buffer.ByteLength(array));
+            Assert.Equal(g, array[g + 50]);
         }
     }
 
     [Fact]
-    public static void TestGetByteAndSetByte()
+    public static unsafe void TestMemoryCopy_OverlappingBuffersSmallCopy()
```

Comment:
Isn't this test identical to the previous one except for a few different numbers? Rather than duplicating it, could it be a theory with different inputs?